### PR TITLE
Remove ConfigInterface::getDir, ConfigInterface::setDir, Finder::setDir and whole FinderInterface

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

**Comment by keradus** [comment]
ref #1763

File: tests/Console/ConfigurationResolverTest.php [hunk 1]
```diff
@@ -246,10 +246,10 @@ public function testResolvePathRelative()
     {
         $this->resolver
             ->setCwd(__DIR__)
-            ->setOption('path', 'Foo'.DIRECTORY_SEPARATOR.'Bar')
+            ->setOption('path', 'Command')
```
Comment (gharlan) [inline tests/Console/ConfigurationResolverTest.php]: why this change?

Comment (keradus) [inline tests/Console/ConfigurationResolverTest.php]: Because now it will call `finder->in(...)`, which would blow up with exception of non-existing `Foo\Bar` dir.

before it was only setting the property (which was later used in `finder->in(...)` call and then throw exception, but utest didn't went so far)

Comment (gharlan) [inline tests/Console/ConfigurationResolverTest.php]: ok

**Comment by gharlan** [comment]
:+1:
